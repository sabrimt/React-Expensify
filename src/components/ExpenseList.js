import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

// const ExpenseList = (props) => (
//     <div>
//         <h1>Expense List</h1>
//         <p><strong>Sort By: </strong>{props.filters.text}</p>
//         {props.expenses.map((expense) => (<p key={expense.description}>{expense.description}</p>))}
//     </div>
// );
//
// const mapStateToProps = (state) => {
//
//     return {
//         expenses: getVisibleExpenses(state.expenses, state.filters),
//         filters: state.filters
//     };
// };


export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map(
                    (expense) => <ExpenseListItem key={expense.id} {...expense} />
                )
            )
        }
        {}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
