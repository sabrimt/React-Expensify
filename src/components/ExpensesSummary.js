import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <h4>
            viewing {props.expenseCount} expense{props.expenseCount !== 1 && 's'} totalling {
                numeral(props.expensesTotal / 100).format('$0,0.00')
            }
        </h4>
    </div>
);

const mapStateToProps = ((state) => ({
    expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(getVisibleExpenses(state.expenses, state.filters))
}));

export default connect(mapStateToProps)(ExpensesSummary);
