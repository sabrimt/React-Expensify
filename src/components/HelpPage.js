import React from 'react';

const HelpPage = () => (
    <div>
        this is from HELP page
    </div>
);

export default HelpPage;
