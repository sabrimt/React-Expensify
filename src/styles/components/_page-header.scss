.page-header {
    background: $off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
}

.page-header__actions {
    margin-top: $l-size;
}

.page-header__title {
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
}

.page-header__sub-title {
    color: $grey;
    font-style: italic;
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 600;
    }
}
