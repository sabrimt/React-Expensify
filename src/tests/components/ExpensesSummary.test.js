import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';


let expenseCount, expensesTotal, wrapper;
beforeEach(() => {
    expenseCount = 1;
    expensesTotal = 9434;
    wrapper = shallow(<ExpensesSummary
        expenseCount={expenseCount}
        expensesTotal={expensesTotal}
    />);
});

test('Should render ExpensesSummary correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should render render ExpensesSummary with 1 expense info', () => {
    expect(wrapper.find('h4').text()).toBe('viewing 1 expense totalling $94.34');
});

test('Should render render ExpensesSummary with 2 expense info', () => {
    wrapper.setProps({expenseCount: 2});
    expect(wrapper.find('h4').text()).toBe('viewing 2 expenses totalling $94.34');
});
