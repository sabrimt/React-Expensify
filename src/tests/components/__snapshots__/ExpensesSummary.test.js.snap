// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly 1`] = `
<div>
  <h4>
    viewing 
    1
     expense
     totalling 
    $94.34
  </h4>
</div>
`;
